docs/standard/serialization/system-text-json/populate-properties.md

```diff
@@ -0,0 +1,111 @@
+dev_langs:
+  - "csharp"
+ms.topic: how-to
+---
+
+# Populate initialized properties
+
+Starting in .NET 8, you can specify a preference to either [replace](#default-replace-behavior) or [populate](#populate-behavior) .NET properties when JSON is deserialized. The <xref:System.Text.Json.Serialization.JsonObjectCreationHandling> enum fields provide the object creation handling choices:
+
+- <xref:System.Text.Json.Serialization.JsonObjectCreationHandling.Replace?displayProperty=nameWithType> (matches the default behavior)
+- <xref:System.Text.Json.Serialization.JsonObjectCreationHandling.Populate?displayProperty=nameWithType>
+
+## Default (replace) behavior
+
+The System.Text.Json deserializer always creates a new instance of the target type. However, even though a new instance is created, some properties and fields might already be initialized as part of the object's construction. Consider the following type:
+
+```csharp
+class A
+{
+    public List<int> Numbers1 { get; } = new List<int>() { 1, 2, 3 };
+    public List<int> Numbers2 { get; set; } = new List<int>() { 1, 2, 3 };
+}
+```
+
+When you create an instance of this class, the `Numbers1` (and `Numbers2`) property's value is a list with three elements (1, 2, and 3). If you deserialize JSON to this type, the *default* behavior is that property values are *replaced*:
+
+- For `Numbers1`, since it's read-only (no setter), it still has the values 1, 2, and 3 in its list.
+- For `Numbers2`, which is read-write, the contents of the list are effectively replaced with the values from the JSON.
+
+For example, if you execute the following deserialization code, `Numbers1` contains the values 1, 2, and 3 and `Numbers2` contains the values 4, 5, and 6.
+
+```csharp
+A? a = JsonSerializer.Deserialize<A>("""{"Numbers1": [4,5,6], "Numbers2": [4,5,6]}""");
+```
+
+## Populate behavior
+
+Starting in .NET 8, you can change the deserialization behavior to modify (*populate*) initialized properties and fields instead of replacing them:
+
+- For a collection type property, any existing values are kept, and new values from the JSON are added to the collection.
+- For a non-collection reference type property, its mutable properties are updated to the JSON value but the reference itself doesn't change.
+- For a struct type property, the effective behavior is that for its mutable properties, the existing values are kept, and new values from the JSON are added. However, unlike a reference property, the object itself isn't reused since it's a value type. Instead, a copy of the struct is modified and then reassigned to the property. For example, after executing the following code, `a.S1.Numbers3` contains the values 1, 2, 3, 4, 5, and 6.
+
+  ```csharp
+  A? a = JsonSerializer.Deserialize<A>("""{"S1": {"Numbers3": [4,5,6]}}""");
+
+  class A
+  {
+    public S S1 { get; set; } = new S();
+  }
+
+  struct S
+  {
+      public S() { }
+      [JsonObjectCreationHandling(JsonObjectCreationHandling.Populate)]
```

Comment:
@krwq @eiriktsarpalis I'm confused as to why I needed to put this attribute on the _struct's_ property instead of on the A.S1 property in order to get population to work. Is that the expected behavior?

Reply:
Yes. The attribute is scoped to the particular property (or contained properties in the case of type-level annotations) but cannot apply transitively to members of the property type. 

In other words, applying the attribute at `A.S1` would only have effect if `S1` itself didn't have a setter.